File: output/CodeGen/codenet/C++/Java/s950360470.java
public class s950360470 {
    public int modPow(int a, int n, int mod) {
        if (n == 0) return 1;
        int res = 1;
        while (n > 0) {
            if ((n & 1) == 1) res = (res * a) % mod;
            a = (a * a) % mod;
            n >>= 1;
        }
        return res;
    }
    public int modInv(int a, int mod) {
        return modPow(a, mod - 2, mod);
    }
    public int modNChooseK(int n, int k) {
        if (n == 0 || k == 0) return 1;
        int p = 1, q = 1;
        for (int i = n - k + 1; i <= n; i++)
            p = (p * i) % MOD;
        for (int i = 2; i <= k; i++)
            q = (q * i) % MOD;
        q = modInv(q, MOD);
        return (p * q) % MOD;
    }
}

File: output/CodeGen/codenet/C++/Python/s950360470.py
def mod_nCk(n, k, mod):
    if n == 0 or k == 0:
        return 1
    p = 1
    q = 1
    for i in range(n - k + 1, n + 1):
        p = (p * i) % mod
    for i in range(2, k + 1):
        q = (q * i) % mod
    q = mod_inv(q, mod)
    return (p * q) % mod

def mod_inv(a, mod):
    if a == 0:
        return 0
    return mod_pow(a, mod - 2, mod)

def mod_pow(a, n, mod):
    if n == 0:
        return 1
    if n == 1:
        return a
    if n % 2 == 0:
        return mod_pow(a, n / 2, mod) * mod_pow(a, n / 2, mod) % mod
    else:
        return mod_pow(a, n / 2, mod) * mod_pow(a, n / 2, mod) * a % mod

n = int(raw_input())
k = int(raw_input())
mod = int(raw_input())
print mod_nCk(k, n, mod)